android/app/src/main/java/app/organicmaps/routing/NavigationController.java

```diff
@@ -260,4 +271,12 @@ public void onStopClicked()
     RoutingController.get().cancel();
   }
 
+  private void updateSpeedLimit(@NonNull final RoutingInfo info)
+  {
+    final Location location = LocationHelper.from(mFrame.getContext()).getSavedLocation();
+    if (location == null)
+      return;
+    mSpeedLimit.setCurrentSpeed(location.getSpeed());
+    mSpeedLimit.setSpeedLimit(info.speedLimitMps);
```

Comment:
I'd say this is not a correct implementation.
I bet there are no people in the whole world who prefer `mps` (meters per second) over `kph` or `mph`.
We should display platform units here.

Please, check the implementation in the original PR: #7005 
I think it's better to rewrite the RoutingInfo class to have `speedLimit` (in platform units), not `speedLimitMps`

Because right now it looks really weird:
* Pass value from C++ (core) to Java (platform) in the core format
* From Java call C++ func to transform core format into platform's

And we don't even have the function to convert this value 😁 
We only have a function to convert `double Mps` into `String platformUnits`

Why not pass the correct format from the beginning?

Reply:
This only calls this function:

```
 public void setSpeedLimit(final double speedLimit)
  {
    if (mSpeedLimit == speedLimit)
      return;

    mSpeedLimit = speedLimit;
    if (mSpeedLimit <= 0)
    {
      mSpeedLimitStr = null;
      setVisibility(GONE);
      return;
    }

    Pair<String, String> speedLimitAndUnits = StringUtils.nativeFormatSpeedAndUnits(mSpeedLimit);
    setVisibility(VISIBLE);
    mSpeedLimitStr = speedLimitAndUnits.first;
    configureTextSize();
    invalidate();
  }
```
And here it is transformed to km/h or mp/h